ingestion/tests/unit/profiler/test_metrics.py

```diff
@@ -111,3 +125,72 @@ def test_table_count(self):
         profiler = SingleProfiler(table_count, session=self.session, table=User)
         res = profiler.execute()
         assert res.get(Metrics.ROW_NUMBER.name) == 2
+
+    def test_avg(self):
+        """
+        Check avg for distinct types
+        """
+
+        # Integer
+        avg = Metrics.AVG(col=User.age)
+        res = SingleProfiler(avg, session=self.session, table=User).execute()
+
+        assert res["AVG"] == 30.5
+
+        # String
+        avg = Metrics.AVG(col=User.name)
+        res = SingleProfiler(avg, session=self.session, table=User).execute()
+
+        assert res["AVG"] == 4.0
+
+        # Text
+        avg = Metrics.AVG(col=User.comments)
+        res = SingleProfiler(avg, session=self.session, table=User).execute()
+
+        assert res["AVG"] == 15.0
+
+    def test_distinct(self):
+        """
+        Check distinct count
+        """
+        dist = Metrics.DISTINCT(col=User.age)
+        res = SingleProfiler(dist, session=self.session, table=User).execute()
+
+        assert res["DISTINCT"] == 2
+
+    def test_duplicate_count(self):
+        """
+        Check composed duplicate count
+        """
+        count = Metrics.COUNT(col=User.name)
+        dist = Metrics.DISTINCT(col=User.name)
+        dup_count = Metrics.DUPLICATE_COUNT(col=User.name)
+        res = SingleProfiler(
+            count, dist, dup_count, session=self.session, table=User
+        ).execute()
+
+        assert res["DUPLICATECOUNT"] == 0
+
+    def test_histogram(self):
+        """
+        Check histogram computation
+        """
+
+        # Cook some data first
+        class TestHist(Base):
+            __tablename__ = "test_hist"
+            id = Column(Integer, primary_key=True)
+            num = Column(Integer)
+
+        TestHist.__table__.create(bind=self.engine)
+
+        data = [TestHist(num=int(rand)) for rand in normal(loc=0, scale=10, size=2000)]
+
+        self.session.add_all(data)
+        self.session.commit()
+
+        hist = Metrics.HISTOGRAM(TestHist.num, bins=5)
```

Comment:
We can now pass kwargs when creating metrics to be more flexible in internal functionalities. Still needs to be updated in the Processor